core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/WebViewYouTubePlayer.kt

```diff
@@ -66,6 +83,46 @@ private class YouTubePlayerImpl(private val webView: WebView) : YouTubePlayer {
     }
     mainThread.post { loadUrl("javascript:$function(${stringArgs.joinToString(",")})") }
   }
+
+  /**
+   * Executes JavaScript and gets the return value via a callback.
+   *
+   * @param javascript The JavaScript code to execute
+   * @param callback Function that will be called with the result
+   */
+  private fun WebView.evaluateWithResult(javascript: String, callback: (String) -> Unit) {
```

Comment:
There are a few issues with this:

1. 
`evaluateJavascript` seem to cause the player to glitch and skip some frames.

2.
`addJavascriptInterface` doesn't work. The page needs to be re-loaded after this is added.

I think unfortunately a different approach might be required.

1. We should leverage the existing `YouTubePlayerBridge` to receive data from JS.
2. Data intended as return value for a method invocation like `isMute` should be place into a queue.
3. When methods like `isMute` are invoked, they should poll the queue until the data they're looking for becomes available, or timeout.